src/code.c

```diff
@@ -259,6 +259,10 @@ Stat NewStatOrExpr(CodeState * cs, UInt type, UInt size, UInt line)
     StatHeader * header = STAT_HEADER(cs, stat);
     header->line = line;
     header->size = size;
+    // check size fits inside header
+    if (header->size != size) {
+        Panic("Function too large");
```

Comment:
Why a panic (immediate abort of the process) and not a simple error?

Reply:
I put a panic in first, mainly to check I've got the right place.

An error might be OK, but looking at code.c, there aren't currently any errors there, so I wasn't sure what would happen if, inside coding, we caused an error -- is GAP in a fit state to produce an error when part way through coding?

Reply:
Yes, calling ErrorQuit is perfectly fine. It might even be able to add a test case exercising that?

Reply:
Actually, this was good, as it's easier to test with ErrorQuit. I've added two tests, both to catch maximum list size works, and one larger produces the error.